eth/contracts/MostL2.sol

```diff
@@ -16,12 +27,160 @@ contract MostL2 is AbstractMost {
+        bytes32 destTokenAddress,
+        uint256 amount,
+        bytes32 destReceiverAddress
+    ) internal override {
+        processedRequests[requestHash] = true;
+        delete pendingRequests[requestHash];
+
+        address _destTokenAddress = bytes32ToAddress(destTokenAddress);
+        address _destReceiverAddress = bytes32ToAddress(destReceiverAddress);
+
+        require(
+            _destTokenAddress != address(0),
+            "We dont bridge native from L2"
+        );
+        require(
+            !isLocalToken[_destTokenAddress],
+            "We dont bridge non local token"
+        );
+
+        // transfer native
+        if (_destTokenAddress == bAzeroAddress) {
+            // So what we do here is:
+            // 1. Mint `amount` Bazero
+            // 2. Allow spending that many Bazero for swap contract
+            // 3. exchange bazero for native tokens, here the swap spends its allowance and sends native to this contract
+            // 4. transfer exchanged native to the receiver.
+            IWrappedToken bazero = IWrappedToken(_destTokenAddress);
+            bazero.mint(address(this), amount);
+
+            (bool swapSuccess, uint256 amount_out) = swap_from_bazero(amount);
+
+            if (!swapSuccess) {
+                emit SwapFailed(requestHash, amount);
+                return;
+            }
+
+            // payout to receiver
+            (bool sendNativeEthSuccess, ) = _destReceiverAddress.call{
+                value: amount_out,
+                gas: GAS_LIMIT
+            }("");
+
+            if (!sendNativeEthSuccess) {
+                emit NativeTransferFailed(requestHash);
+            } else {
+                emit NativeTransferSwap(requestHash, amount_out);
+            }
+            emit RequestProcessed(requestHash);
+        } else {
+            // TODO non native transfer
+        }
+    }
+
+    function burn_bazero(uint256 amount) internal {
+        IWrappedToken bazero = IWrappedToken(bAzeroAddress);
+        bazero.burn(amount);
+    }
+
+    /// @notice Invoke this tx to transfer funds to the destination chain.
+    /// Account needs to send native Azero which are swapped for bazero
+    /// tokens.
+    ///
+    /// @dev Tx emits a CrosschainTransferRequest event that the relayers listen to
+    /// & forward to the destination chain.
+    function sendRequestNative(
+        bytes32 destReceiverAddress
+    ) external payable override whenNotPaused {
+        uint256 amount = msg.value;
+        require(amount != 0, "Zero amount");
+        if (amount == 0) revert ZeroAmount();
+        if (destReceiverAddress == bytes32(0)) {
+            revert ZeroAddress();
+        }
+
+        bytes32 destTokenAddress = supportedPairs[
+            addressToBytes32(bAzeroAddress)
+        ];
+
+        if (destTokenAddress == 0x0) {
+            revert UnsupportedPair();
+        }
+
+        uint256 amount_out = swap_for_bazero(amount);
+        burn_bazero(amount_out);
```

Comment:
Can this fail? E.g. some bug happens so that `amount_out` > `total_supply` of bzero, how `burn_azero` behaves in such situation?

Reply:
this can fail in this scenario